mm2src/coins/eth.rs

```diff
@@ -249,6 +249,42 @@ impl From<web3::Error> for BalanceError {
     fn from(e: web3::Error) -> Self { BalanceError::Transport(e.to_string()) }
 }
 
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+#[allow(dead_code)]
+pub enum EthActivationV2Error {
+    InvalidPayload(String),
+    #[display(fmt = "Platform coin {} activation failed. {}", ticker, error)]
+    ActivationFailed {
+        ticker: String,
+        error: String,
+    },
+    CouldNotFetchBalance(String),
+    UnreachableNodes(String),
+    AtLeastOneNodeRequired(String),
+    InternalError(String),
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct EthActivationV2Request {
+    pub nodes: Vec<EthNode>,
+    pub swap_contract_address: Address,
+    pub fallback_swap_contract: Option<Address>,
+    pub gas_station_url: Option<String>,
+    pub gas_station_decimals: Option<u8>,
+    #[serde(default)]
+    pub gas_station_policy: GasStationPricePolicy,
+    pub mm2: Option<u8>,
+    pub tx_history: Option<bool>,
```

Comment:
It can be bool with `#[serde(default)]`.

Reply:
done